src/TypedMap.php

```diff
@@ -22,11 +22,13 @@
      */
     public function __construct(array $input = [], $flags = 0, $iterator_class = "ArrayIterator")
     {
+        parent::__construct([], $flags, $iterator_class);
```

Comment:
Because we want to set the values one by one using the `offsetSet` method in this class, so that the keys get stored properly. And TypedMap accepts objects as keys, whereas ArrayObject does not, so it would break on those. Unfortunately ArrayObject's `__construct` setting of values does not use offsetSet but rather sets its value directly (which makes sense from its perspective).